src/simulation/elements/LIGH.cpp

```diff
@@ -292,9 +295,18 @@ static void create_line_par(Simulation * sim, int x1, int y1, int x2, int y2, in
 static int graphics(GRAPHICS_FUNC_ARGS)
 {
 	*firea = 120;
-	*firer = *colr = 235;
-	*fireg = *colg = 245;
-	*fireb = *colb = 255;
+	if (cpart->dcolour)
+	{
+		*firer = *colr = PIXR(cpart->dcolour);
+		*fireg = *colg = PIXG(cpart->dcolour);
+		*fireb = *colb = PIXB(cpart->dcolour);
+	}
+	else
+	{
+		*firer = *colr = 235;
+		*fireg = *colg = 245;
+		*fireb = *colb = 255;
+	}
 	*pixel_mode |= PMODE_GLOW | FIRE_ADD;
 	return 1;
```

Comment:
A few things aren't correct here.

You added a special case for LIGH into Renderer.cpp. But actually, "isready" is only set by returning 1. If you make a non-static graphics function then you should return 0 instead (and then remove the special case).

But directly setting dcolor is unnecessary, you can add the DECO_FIRE pixel mode to make the deco color apply to fire graphics instead of doing it yourself.